### PR TITLE
Feature

### PR DESCRIPTION
**Pull request by IBUNHABIBU**
-The weather app project is ready. We can able to fetch the temperature data from weathermap api 
and display it to the web browser.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,51 @@
...
+
+## Usage
+Clone this repository 
+ `$npm install` 
+ `$npm run start:dev`
+ 
+
+## [Live demo](https://raw.githack.com/IBUNHABIBU/weather-app/feature/dist/index.html "Of the project") of the project
+
+## link to  [Video presentation](https://#"Loom")
```
Comment (shshamim63) [inline README.md]: You do not have any video presentation. Please remove it.

File: README.md [hunk 2]
```diff
@@ -1 +1,51 @@
...
+* Webpack
+* openweathermap
+
+## Usage
+Clone this repository 
+ `$npm install` 
+ `$npm run start:dev`
+ 
+
+## [Live demo](https://raw.githack.com/IBUNHABIBU/weather-app/feature/dist/index.html "Of the project") of the project
```
Comment (shshamim63) [inline README.md]: Live version is not working.

**Review by shshamim63 — CHANGES_REQUESTED** [review]
## :recycle: Changes Required :recycle: 
___
Hi @IBUNHABIBU :wave: 
Awesome job, I liked the way you have added temperature data, seems eye-catching to me :clap: . But there are some tasks that you need to add to make this project complete. Let's check them.
___
### :spades: Mandatory Changes :spades: 
___
- Please update your readme a bit more. Check the comments.
- It should contain a toggle control to determine if the result will be shown in Celsius or Fahrenheit ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/search_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/weather-app))
- It should change its appearance based on the received data ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/appearance_requirement.png) from [Odin](https://www.theodinproject.com/courses/javascript/lessons/weather-app))
___
:rocket: Just a reminder for the upcoming tasks. Keep the application logic separated from DOM manipulation tasks ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/section/dom_logic_separation_requirement.png) from Pathwright)
___
#### Please resolve the issues mentioned above and send another review request.
___
## :herb: Happy Coding :herb: 
### Shakhawat Hossain

**Comment by IBUNHABIBU** [comment]
I have done the following changes
- Adding the toggling switch to change the temperature from Celsius to Fahrenheit 
- Adding the feature for showing weather status icon like cloudy sunny etc.
- Adding the live link to the readme file

File: src/modules/app.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
+import UI from './user_interface';
+
+import '../css/style.css';
+
+UI();
+const api = {
+  key: 'd79e7d987356c35ab053eb9c2eb96551',
+  base: 'https://api.openweathermap.org/data/2.5/',
+};
```
Comment (shloch) [inline src/modules/app.js:12]: logic code

File: src/modules/app.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  class="checkbox" /> 
+    <label for="switch" class="toggle"> 
+    <p><span class="celcious">C</span><span class="faren">F</span></p> 
+    </label> `;
+}
+function getResult(data) {
+  fetch(`${api.base}weather?q=${data}&units=metric&APPID=${api.key}`)
+    .then(response => response.json())
+    .then(displayResult);
+}
```
Comment (shloch) [inline src/modules/app.js:19]: logic code

**Review by shloch — CHANGES_REQUESTED** [review]
## Hello @IBUNHABIBU , 

Great work, you brought a whole lot of modifications as proposed and it works greatly

weather

### [changes required]

- You however forgot one modification proposed, which is [changing the page look](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/weather_app/appearance_requirement.png) based on the temperature value,
You can do this by basically changing the background image with maybe sun for a hot temperature and a cold-ice background for a cold temperature value for example

- Also, I would add that you should [separate DOM from LOGIC](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/section/dom_logic_separation_requirement.png), main logic in this project is the code to fetch data from API, please put that in a separate file and import it into `app.js`

It's a good thing, you've already got things running correctly...Please let's polish the project
Happy coding

**Comment by IBUNHABIBU** [comment]
- I have seperated the application logic from dom manipulation in a separate file
- I have used icons from openweathermap to show the status of the weathe i.e cloudy with cloud icon and sunny with sun icon

File: src/modules/fetch.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+  const toggleSwitch = document.querySelector('.units');
+  toggleSwitch.innerHTML = `<input type="checkbox" id="switch"
+  class="checkbox" /> 
+    <label for="switch" class="toggle"> 
+    <p><span class="celcious">C</span><span class="faren">F</span></p> 
+    </label> `;
+}
+// function getResult(data) {
+//   fetch(`${api.base}weather?q=${data}&units=metric&APPID=${api.key}`)
+//     .then(response => response.json())
```
Comment (mosaaleb) [inline src/modules/fetch.js]: - [ ] Comments: It’s better to delete commented piece of codes. If you need to review a previous version of your code, use git history .

File: src/modules/app.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+UI();
+const api = {
+  key: 'd79e7d987356c35ab053eb9c2eb96551',
+  base: 'https://api.openweathermap.org/data/2.5/',
+};
+const search = document.querySelector('.searchbox');
+
+function getResult(data) {
+  fetch(`${api.base}weather?q=${data}&units=metric&APPID=${api.key}`)
+    .then(response => response.json())
```
Comment (mosaaleb) [inline src/modules/app.js:17]: - [ ] **[OPTIONAL]** What about adding catch when there is no search results found, and displaying useful information to the user?

**Review by mosaaleb — CHANGES_REQUESTED** [review]
#### Well done @IBUNHABIBU :clap:

Excellent job handling the requirements from the previous review. Yet there is some requirements needs to be met before moving on.

- [ ] The application should change its appearance based on the received data.



Also please check out the comments below.
Please handle these requirements and submit a new code review.

##### Muhammad Ebeid

File: src/modules/app.js (hunk 3 above)
Comment (IBUNHABIBU) [reply src/modules/app.js:12]: -I have implemented the feature for the page to change background according to the weather status

**Review by mcrd25 — APPROVED** [review]
# Changes Approved :heavy_check_mark: 

Hi @IBUNHABIBU :wave: 

Good job, no need for further review! :smile: 

***If you have any questions, now or in the future, do not hesitate to contact me on social media, email or on slack and I will be glad to chat with you.***

*Happy coding 😄,*
**[--Maya](https://github.com/mcrd25)**
[GitHub](https://github.com/mcrd25) [Twitter](https://twitter.com/mcrd25) [linkedin](https://www.linkedin.com/in/mayadouglas/) [GitLab](https://gitlab.com/mcrd25)

**Pull request merged**